### ISSUE TITLE
EDD_Payment::update_status() should optionally return EDD_Payment

### ISSUE THREAD

**Issue by Shelob9** [easydigitaldownloads/easy-digital-downloads#5310]
When I use EDD_Payment::update_status() after  I update the status, I then have to recreate the object.

This explains the problem:
```
	$payment->update_status( 'publish' );
	$_payment = new \EDD_Payment( $payment->ID );
	echo $payment->status; //pending
	echo $_payment->status; //publish
```

As a result, I'm doing this:
```
	$payment->update_status( 'publish' );
	$payment = new \EDD_Payment( $payment->ID );
```
which is silly.

lf instead the method  returned the modified object (would have to be a non-default option for backwards compat reasons), I would just do `$payment->update_status( 'publish' );` and be fine.

I'm not submitting a PR yet, beacuse if @pippinsplugins etc agree with me, I'll argue that this whole class should use a fluent interface. save(), add_download(), etc. should all return $this :)

**Comment by Shelob9** [comment]
Ok, I also realize that `EDD_Payment::update_status()` also doesn't update the object properties either.

Here is what I am using instead:
```
	namespace calderawp\cfedd\edd\payment;
	
	
	class payment extends \EDD_Payment {
	
		/**
		 * @inheritdoc
		 * @return payment
		 */
		public function save() {
			parent::save();
			return $this;
	
		}
	
		/**
		 * @inheritdoc
		 * @return payment
		 */
		public function update_status( $status = false ) {
			if( parent::update_status( $status ) ){
				$this->status = $status;
				$this->post_status = get_post_status( $this->ID );
			}
	
			return $this;
	
		}
	
	}
```

I *think* but am not sure yet that I'm going to propose a class like this that does similar for every method, or does something fancy with `__call()`.

**Comment by pippinsplugins** [comment]
Thanks for the report @Shelob9, we'll get that fixed!

**Comment by cklosowski** [comment]
This should hopefully be fixed by #4718 as it will require us to update the properties when we make changes.
